### PR TITLE
Allow zero velocity to be valid ( test fixes )

### PR DESCRIPTION
**Pull request by gimait**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1642, (Original PR #1792) |
| Primary OS tested on | (Ubuntu 18.04) |

---

## Description of contribution in a few bullet points

This PR includes the changes in PR #1792 plus a small bug fix (which was failing some tests) and corrections to the style.
The main contribution (by @gokul-gokz) included the following points:

* Removed the zero velocity check from kinematic parameters. This allows critics/users to have zero velocities as valid.
* Moved the zero velocity check to one more level higher(xy_theta_iterator). This plugin can be overridden if needed.

### CODE REVIEW

**Comment by SteveMacenski** [comment]
@bitfroest please review and let me know if this fixes your concerns.

File: nav2_dwb_controller/dwb_plugins/include/dwb_plugins/xy_theta_iterator.hpp [hunk 1]
```diff
@@ -58,6 +58,23 @@ class XYThetaIterator : public VelocityIterator
...
+   * @return True if the magnitude hypot(x,y) and theta are within the robot's absolute limits
+   *
+   * This is based on three parameters: min_speed_xy, max_speed_xy and min_speed_theta.
+   * The speed is valid if
+   *  1) The combined magnitude hypot(x,y) is less than max_speed_xy (or max_speed_xy is negative)
+   *  AND
+   *  2) min_speed_xy is negative or min_speed_theta is negative or
+   *     hypot(x,y) is greater than min_speed_xy or fabs(theta) is greater than min_speed_theta.
+   *
+   * In English, it makes sure the diagonal motion is not too fast,
```
Comment (SteveMacenski) [inline nav2_dwb_controller/dwb_plugins/include/dwb_plugins/xy_theta_iterator.hpp:72]: Please remove the english and latin bits. Its funny, I like it, but TMI

File: nav2_dwb_controller/dwb_plugins/src/xy_theta_iterator.cpp [hunk 2]
```diff
@@ -85,9 +92,25 @@ void XYThetaIterator::startNewIteration(
...
+  double vmag_sq = x * x + y * y;
+  if (kinematics.getMaxSpeedXY() >= 0.0 && vmag_sq > kinematics.getMaxSpeedXY_SQ() + EPSILON) {
+    return false;
+  }
+  if (kinematics.getMinSpeedXY() >= 0.0 && vmag_sq + EPSILON < kinematics.getMinSpeedXY_SQ() &&
+    kinematics.getMinSpeedTheta() >= 0.0 && fabs(theta) + EPSILON < kinematics.getMinSpeedTheta())
+  {
+    return false;
+  }
+  if (vmag_sq == 0.0 && th_it_->getVelocity() == 0.0) {return false;}
```
Comment (SteveMacenski) [inline nav2_dwb_controller/dwb_plugins/src/xy_theta_iterator.cpp:107]: Please expand the return false onto new lines

**Comment by Bitfroest** [comment]
LGTM! Closes #1642 
Thanks @gimait @SteveMacenski

File: nav2_dwb_controller/dwb_plugins/include/dwb_plugins/xy_theta_iterator.hpp (hunk 1 above)
Comment (gimait) [reply nav2_dwb_controller/dwb_plugins/include/dwb_plugins/xy_theta_iterator.hpp:72]: hahaha for sure

**Comment by SteveMacenski** [comment]
@gimait LMK when this passes CI and I'll merge

**Pull request merged**